File: Robot/RobotControl/app/src/main/java/ec/edu/espe/robotcontrol/ConfigurationControl/ConfigFileManager.kt
package ec.edu.espe.robotcontrol.ConfigurationControl


import android.os.Environment
import com.github.kittinunf.fuel.Fuel
import com.github.kittinunf.fuel.core.extensions.jsonBody
import java.io.File
import java.util.*


fun writeRobotNameFile(robotName: String,configJSON:String){
    val configFile = File(Environment.getExternalStorageDirectory(),"/Config/config.txt")
    configFile.printWriter().use { out ->
        out.println(robotName)
        out.println("@@@")
        out.println(configJSON)
    }
}

fun writeRobotMapFile(robotName: String,mapJSON:String){
    val mapFile = File(Environment.getExternalStorageDirectory(),"/Config/map.txt")
    mapFile.printWriter().use { out ->
        out.println(robotName)
        out.println("@@@")
        out.println(mapJSON)
    }
}

fun readConfigFile():String{
    var dataString: String = ""
    println(Environment.getExternalStorageDirectory())
    File(Environment.getExternalStorageDirectory(),"/Config/config.txt").forEachLine {
        dataString += it
        println(it)
    }
    return dataString;
}
fun readMapFile():String{
    var dataString: String = ""
    println(Environment.getExternalStorageDirectory())
    File(Environment.getExternalStorageDirectory(),"/Config/map.txt").forEachLine {
        dataString += it
        println(it)
    }
    return dataString;
}

fun postConfigFile(robotName:String, configString:String){
    Fuel.post("https://us-central1-tsrrcontrolmonit.cloudfunctions.net/addNewRobot?robot="  + robotName)
        .jsonBody(configString)
        .also { println(it) }
        .response { result ->
            println(result)
        }
}
fun postMapFile(robotName:String, mapString:String){
    Fuel.post("https://us-central1-tsrrcontrolmonit.cloudfunctions.net/addNewRobotMapping?robot="  + robotName)
        .jsonBody(mapString)
        .also { println(it) }
        .response { result ->
            println(result)
        }
}

fun robotSetup(): String{//returns the robot generated name

    var configString = readConfigFile()
    var mapString = readMapFile()
    var robotName = configString.split("@@@")[0]
    var configJSON = configString.split("@@@")[1]
    var robotMapName = mapString.split("@@@")[0]
    var mapJSON = mapString.split("@@@")[1]
    if(robotName.equals("none")){
        val uuid = UUID.randomUUID()
        robotName = "robot" + uuid.toString()
        postConfigFile(robotName,configJSON)
        writeRobotNameFile(robotName,configJSON)
    }
    if(robotMapName.equals("none")){
        robotMapName = robotName
        postMapFile(robotMapName,mapJSON)
        writeRobotMapFile(robotMapName,mapJSON)
    }
    return robotName
}

File: Robot/RobotControlPython/src/ConfigurationControll/ConfigFileManager.py
import uuid
import requests
import json

def writeRobotNameFile(robotName,configJSON):
    configFile = open("/home/pi/RobotControlPython/src/Config.txt","w+")
    configFile.write(robotName)
    configFile.write("@@@")
    configFile.write(configJSON)
    configFile.close()

def writeRobotMapFile(robotName,mapJSON):
    configFile = open("/home/pi/RobotControlPython/src/Map.txt","w+")
    configFile.write(robotName)
    configFile.write("@@@")
    configFile.write(mapJSON)
    configFile.close()

def readConfigFile():
    fileContents = ""
    configFile = open("/home/pi/RobotControlPython/src/Config.txt","r")
    if(configFile.mode == "r"):
        fileContents = configFile.read()
        print(fileContents)
    return fileContents

def readMapFile():
    fileContents = ""
    configFile = open("/home/pi/RobotControlPython/src/Map.txt","r")
    if(configFile.mode == "r"):
        fileContents = configFile.read()
        print(fileContents)
    return fileContents

def postConfigFile(robotName, configJSON):
    response = requests.post("https://us-central1-tsrrcontrolmonit.cloudfunctions.net/addNewRobot?robot="  + robotName, json= json.loads(configJSON))
    print(response.status_code, response.reason)

def postMapFile(robotName, mapJSON):
    response = requests.post("https://us-central1-tsrrcontrolmonit.cloudfunctions.net/addNewRobotMapping?robot="  + robotName, json= json.loads(mapJSON))
    print(response.status_code, response.reason)

def robotSetup():
    configFileData = readConfigFile().split("@@@")
    mapFileData = readMapFile().split("@@@")
    robotName = configFileData[0]
    configJSON = configFileData[1]
    robotMapName = mapFileData[0]
    mapJSON = mapFileData[1]
    if(robotName == "none\n" or robotName == "none" ):
        robotName =  "robot" + str(uuid.uuid1())
        postConfigFile(robotName,configJSON)
        writeRobotNameFile(robotName,configJSON)
    if(robotMapName == "none\n" or robotMapName == "none" ):
        robotMapName =  robotName
        postMapFile(robotMapName,mapJSON)
        writeRobotNameFile(robotMapName,mapJSON)

    return robotName
